main/src/main/java/SimpleNotePad.java

```diff
@@ -0,0 +1,344 @@
+
+        return action;
+    }
+
+    // Save action listener
+    private ActionListener actionMenuSave() {
+        ActionListener action = new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                File fileToWrite = null;
+                JFileChooser fc = new JFileChooser();
+                int returnVal = fc.showSaveDialog(null);
+                if (returnVal == JFileChooser.APPROVE_OPTION)
+                    fileToWrite = fc.getSelectedFile();
+                try {
+                    PrintWriter out = new PrintWriter(new FileWriter(fileToWrite));
+                    out.println(uiTextPane.getText());
+                    JOptionPane.showMessageDialog(null, "File is saved successfully...");
+                    out.close();
+                    UpdateRecents(fileToWrite.getName());
+                } catch (IOException ex) {
+                    JOptionPane.showMessageDialog(null,
+                            "Save error " + ex, "Save error",
+                            JOptionPane.ERROR_MESSAGE);
+                }
+            }
+        };
+
+        return action;
+    }
+
+    // Print action listener
+    private ActionListener actionMenuPrint() {
+        ActionListener action = new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                try{
+                    PrinterJob pjob = PrinterJob.getPrinterJob();
+                    pjob.setJobName("Sample Command Pattern");
+                    pjob.setCopies(1);
+                    pjob.setPrintable(new Printable() {
+                        public int print(Graphics pg, PageFormat pf, int pageNum) {
+                            if (pageNum>0)
+                                return Printable.NO_SUCH_PAGE;
+                            pg.drawString(uiTextPane.getText(), 500, 500);
+                            paint(pg);
+                            return Printable.PAGE_EXISTS;
+                        }
+                    });
+
+                    if (pjob.printDialog() == false)
+                        return;
+                    pjob.print();
+                } catch (PrinterException pe) {
+                    JOptionPane.showMessageDialog(null,
+                            "Printer error " + pe, "Printing error",
+                            JOptionPane.ERROR_MESSAGE);
+                }
+            }
+        };
+
+        return action;
+    }
+
+    // Copy action listener
+    private ActionListener actionMenuCopy() {
+        ActionListener action = new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                uiTextPane.copy();
+            }
+        };
+
+        return action;
+    }
+
+    // Paste action listener
+    private ActionListener actionMenuPaste() {
+        ActionListener action = new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                StyledDocument doc = uiTextPane.getStyledDocument();
```

Comment:
For paste, this can be boiled down to a single line of code:

JTextPane.paste();